### PR TITLE
Add more crops tag (Cocoa Beans, Cactus, Melon, Pumpkin, and Sugar Cane)

### PR DESCRIPTION
**Pull request by TelepathicGrunt**
Makes sense to me to add for completeness

matches https://github.com/FabricMC/fabric/pull/4008

### CODE REVIEW

**Comment by KnightMiner** [comment]
One small concern is calling the item a crop might suggest the block form should also be a crop, which conflicts with the standard definition of a crop growing on a block below (that is, the class `CropBlock`). Cocoa beans sprout from the side so I half wonder if they shouldnt have a different name so the block tags can mirror this item tag.

**Comment by TelepathicGrunt** [comment]
@KnightMiner the argument I gotten against that is that modded crops may not extend CropBlock but still are a crop. We also have instanceof if we need to explicitly check for CropBlock for specific stuff regarding CropBlock itself. These tags may be better to handle as gameplay behavior than underlying inheritance classes. 

Note to self, add cocoa bean tag to parent tag. forgot to do it here

**Comment by TelepathicGrunt** [comment]
Also knight, there is no `c` crop block tag. So there's nothing to mirror. The vanilla crops tag is a separate thing and is only for determine what plants bees can grow. Nether wart is not in the vanilla crops tag but we consider it a crop for our `c` item crop tag already cause, it's really a crop you grow and harvest. Just mojang didn't completely fill their crop tag since they were only gathering stuff for what plants bees can grow which they won't let nether wart nor cocoa beans be growable by bees.

**Comment by TelepathicGrunt** [comment]
More crop tags added after discussing with some fabric folks since we realized the crops tag wasn't as complete as it should be

File: src/main/java/net/neoforged/neoforge/common/Tags.java [hunk 1]
```diff
@@ -315,11 +315,20 @@ public static class Items {
...
          */
         public static final TagKey<Item> CLUSTERS = tag("clusters");
+        /**
+         * For raw materials harvested from growable plants. Crop items can be edible like carrots or
+         * non-edible like wheat and cocoa beans.
+         */
         public static final TagKey<Item> CROPS = tag("crops");
         public static final TagKey<Item> CROPS_BEETROOT = tag("crops/beetroot");
         public static final TagKey<Item> CROPS_CARROT = tag("crops/carrot");
+        public static final TagKey<Item> CROPS_CACTUS = tag("crops/cactus");
```
Comment (Technici4n) [inline src/main/java/net/neoforged/neoforge/common/Tags.java:324]: Should go above carrot.

File: src/main/java/net/neoforged/neoforge/common/data/internal/NeoForgeItemTagsProvider.java [hunk 2]
```diff
@@ -66,11 +66,20 @@ public void addTags(HolderLookup.Provider lookupProvider) {
                 .add(Items.CYAN_CONCRETE_POWDER).add(Items.PURPLE_CONCRETE_POWDER).add(Items.BLUE_CONCRETE_POWDER)
                 .add(Items.BROWN_CONCRETE_POWDER).add(Items.GREEN_CONCRETE_POWDER).add(Items.RED_CONCRETE_POWDER)
                 .add(Items.BLACK_CONCRETE_POWDER);
-        tag(Tags.Items.CROPS).addTags(Tags.Items.CROPS_BEETROOT, Tags.Items.CROPS_CARROT, Tags.Items.CROPS_NETHER_WART, Tags.Items.CROPS_POTATO, Tags.Items.CROPS_WHEAT);
+        tag(Tags.Items.CROPS).addTags(
+                Tags.Items.CROPS_BEETROOT, Tags.Items.CROPS_CARROT, Tags.Items.CROPS_CACTUS,
```
Comment (Technici4n) [inline src/main/java/net/neoforged/neoforge/common/data/internal/NeoForgeItemTagsProvider.java]: Carrot should go after cactus?

**Review by Technici4n — APPROVED** [review]

**Pull request merged**